### ISSUE TITLE
Change DDL generation such that it does create index before alter table add foreign key

### ISSUE THREAD

**Issue by rbygrave** [ebean-orm/ebean#1336]
## Actual behavior

Returned error : 
pq: foreign key requires an existing index on columns ("identity_provider_id")

----
As per discussion:

### cockroach db Documentation
Documentation about FK / Index: https://www.cockroachlabs.com/docs/stable/foreign-key.html 

Foreign key columns must be indexed. This is required because updates and deletes on the referenced table will need to search the referencing table for any matching records to ensure those operations would not violate existing references 

To meet this requirement when adding the Foreign Key constraint to an existing table, if the columns you want to constraint are not already indexed, use CREATE INDEX to index them and only then use the ADD CONSTRAINT statement to add the Foreign Key constraint to the columns

...

### MySql Documentation

MySQL requires indexes on foreign keys and referenced keys so that foreign key checks can be fast and not require a table scan. In the referencing table, there must be an index where the foreign key columns are listed as the first columns in the same order. Such an index is created on the referencing table automatically if it does not exist. This index might be silently dropped later, if you create another index that can be used to enforce the foreign key constraint. index_name, if given, is used as described previously

So in summary we  should change the order of the statements in the DDL creating the index before adding the foreign key constraint.
